pkg/goldpinger/k8s.go

```diff
@@ -56,9 +108,29 @@ func GetAllPods() map[string]*GoldpingerPod {
 	for _, pod := range pods.Items {
 		podMap[pod.Name] = &GoldpingerPod{
 			Name:   pod.Name,
-			PodIP:  pod.Status.PodIP,
-			HostIP: pod.Status.HostIP,
+			PodIP:  getPodIP(pod),
+			HostIP: getHostIP(pod),
 		}
 	}
 	return podMap
 }
+
+// ipMatchesConfig checks if the input IP family matches the first entry in the IPVersions config.
+// TODO update to check all config versions to support dual-stack pinging.
+func ipMatchesConfig(ip string) bool {
+	ipFamily := getIPFamily(ip)
+	return GoldpingerConfig.IPVersions[0] == ipFamily
+}
+
+// getIPFamily returns the IP family of the input IP.
+// Possible values are 4 and 6.
+func getIPFamily(ip string) string {
+	if k8snet.IsIPv4String(ip) {
+		return string(k8snet.IPv4)
+	}
+	if k8snet.IsIPv6String(ip) {
+		return k8snet.IPv6
```

Comment:
Curious why this one doesn't need string conversion?

Reply:
Or, in fact, looks like the string on line 129 should go instead https://github.com/kubernetes/utils/blob/4a2b36d8d73f0bfa1092e740605b66db34fca1cd/net/port.go#L31

Reply:
I'm not entirely sure why, but the k8s net package only has the IPv4 defined as the IPFamily type, while IPv6 is just a string. So dealing with IPFamily and string will require some conversion somewhere (e.g. if the return type of getIPFamily is string I don't think returning net.IPv4 is sufficient, `Cannot use 'k8snet.IPv4' (type IPFamily) as the type string`).

To make it more readable, I changed the return type as IPFamily and then the caller can convert the result to a string. Other option is to define an IPFamily type in Goldpinger. Let me know what you think.

Reply:
Curious. Sorry for confusion, must have look in the wrong place.

Reply:
@seeker89 you looked in the right place, but there's a subtle (probably) bug there probably due to someone not paying attention in their group const declarations thinking that typing the first declaration types the rest. IPv6 needs to be explicitly typed IPFamily, but since that's just an alias to string I can see how they wouldn't notice during typical use (at least, doing comparisons to string is not a compiler error and that seems to be how that type is used internally).